test/persistence-hooks.suite.js

```diff
@@ -1828,6 +1912,109 @@ module.exports = function(dataSource, should) {
       });
     });
 
+    describe('PersistedModel.replaceOrCreate', function() {
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+
+        TestModel.replaceOrCreate(
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            triggered.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save'
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on replace', function(done) {
+        monitorHookExecution();
+
+        TestModel.replaceOrCreate(
+          { id: existingInstance.id, name: 'new name' },
+          function(err, record, created) {
+            if (err) return done(err);
+            if (dataSource.connector.replaceOrCreate) {
+              triggered.should.eql([
+                'access',
+                'before save',
+                'persist',
+                'loaded',
+                'after save'
+              ]); 
+            } else {
+              triggered.should.eql([
+                'access',
+                'loaded',
+                'before save',
+                'persist',
+                'loaded',
+                'after save'
+              ]);
+            };
+            // TODO: Check order if not implemented in the connector...
+            done();
+          });
+      });
+
+      it('triggers `access` hook', function(done) {
+        TestModel.observe('before save', pushContextAndNext());
+        TestModel.replaceOrCreate({id: existingInstance.id, name: 'new name'},
+        function(err, instance) {
+          if (err)
+            return done(err);
+          observedContexts.should.eql(aTestModelCtx({
+            instance: instance
+          }));
+          done();
+        });
+      });
```

Comment:
> Yes, that's correct. Please consider changing the copied test cases so that they would fail if run against updateOrCreate instead of replaceOrCreate.
TBH, this is the only test which fails against `updateOrCreate` but passes for `replaceOrCreate`.